packages/rpc-methods/src/restricted/manageState.ts

```diff
@@ -169,3 +147,68 @@ function getManageStateImplementation({
     }
   };
 }
+
+/**
+ * Validates the manageState method `params` and returns them cast to the correct
+ * type. Throws if validation fails.
+ *
+ * @param params - The unvalidated params object from the method request.
+ * @param method - RPC method name used for debugging errors.
+ * @param storageSizeLimit - Maximum allowed size (in bytes) of a new state object.
+ * @returns The validated method parameter object.
+ */
+export function getValidatedParams(
+  params: unknown,
+  method: string,
+  storageSizeLimit: number,
```

Comment:
`storageSizeLimit` could default to `STORAGE_SIZE_LIMIT` to simplify all calls to it

Reply:
Changed and the rest is refactored 👍